src/facade/dragonfly_connection.cc

```diff
@@ -736,7 +736,16 @@ auto Connection::ParseMemcache() -> ParserStatus {
     if (MemcacheParser::IsStoreCmd(cmd.type)) {
       total_len += cmd.bytes_len + 2;
       if (io_buf_.InputLen() >= total_len) {
-        value = str.substr(consumed, cmd.bytes_len);
+        std::string_view parsed_value = str.substr(consumed, cmd.bytes_len + 2);
+        if (parsed_value[cmd.bytes_len] != '\r' && parsed_value[cmd.bytes_len + 1] != '\n') {
+          builder->SendClientError("bad data chunk");
+          // We consume the whole buffer because we don't really know where it ends
+          // since the value length exceeds the cmd.bytes_len.
+          io_buf_.ConsumeInput(io_buf_.InputLen());
+          return OK;
```

Comment:
why do you return OK?

Reply:
Because that allows us to continue in the `IoLoop`. The same is done with `PARSE_ERROR`, `BAD_DELTA` etc... We always return `OK`.

Reply:
well, BAD_DELTA is different because we stay in a state machine that allows to continue parsing because your state machine is synchronized with the stream it parses.
if bytes is inconsistent with line endings I do not see how you can recover the stream parsing from this point.

Reply:
> if bytes is inconsistent with line endings I do not see how you can recover the stream parsing from this point.

We drop the payload (via ConsumeInput) and restart. This is what memcached is doing. Try with telnet:

```
telnet set my 0 0 4
kostas
CLIENT_ERROR bad data chunk
set my 0 0 4
kost
STORED
```

You see we don't drop any connection here after the errornous out of range input `kostas`.

> I do not see how you can recover the stream parsing from this point

You start again from scratch, (that's why I consume the io_buf).

What am I missing?